src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_constants.py

```diff
@@ -8,3 +8,6 @@
 STORAGE_RESOURCE_TYPE = 'Microsoft.Storage/storageAccounts'
 WEBHOOK_RESOURCE_TYPE = ACR_RESOURCE_TYPE + '/webhooks'
 WEBHOOK_API_VERSION = '2017-06-01-preview'
+MANAGED_REGISTRY_API_VERSION = '2017-06-01-preview'
+MANAGED_REGISTRY_SKU = ['Managed_Basic', 'Managed_Standard', 'Managed_Premium']
+MANAGED_REGISTRY_LOCATION = ['westcentralus', 'eastus', 'westeurope']
```

Comment:
@tjprescott @derekbekoe This is for sure a bad design/implementation pattern. If we don't hard code the 3 locations, it seems impossible for users to find out which location is available to create a managed registry from the error message returned by ARM:
```Deployment failed. {
  "error": {
    "code": "NoRegisteredProviderFound",
    "message": "No registered resource provider found for location 'westus' and API version '2017-06-01-preview' for type 'registries'. The supported api-versions are '2016-06-27-preview, 2017-03-01, 2017-06-01-preview, 2016-06-27-beta'. The supported locations are 'westus, eastus, southcentralus, westeurope, northeurope, uksouth, ukwest, australiaeast, australiasoutheast, centralindia, eastasia, japaneast, japanwest, southeastasia, southindia, brazilsouth, canadaeast, canadacentral, centralus, eastus2, northcentralus, westcentralus, westus2, eastus2euap, centraluseuap'."
  }
}
```
What is the recommended approach for this scenario? /cc @SajayAntony